File: 49-Group-Anagrams.py
class Solution():
	def groupAnagrams(self, strs):
		d = {}
		grouped = []
		i = 0
		for word in strs:
			k = ''.join(sorted(word))
			if k in d:
				grouped[d[k]].append(word)
			else:
				grouped.append([])
				grouped[i].append(word)
				d[k] = i
				i+=1
		return(grouped)

driver = Solution()
print('\n')

print('1')
print('**',driver.group(['eat','tea','tan','ate','nat','bat']))
print('-- [["ate", "eat", "tea"], ["nat", "tan"], ["bat"]]')

File: 49-Group-Anagrams.cpp
#include <iostream>
#include <map>
#include <vector>

// Problem:
// Given an array of strings
// Return all anagrams group together

// Notes
// All inputs will be in lowercase.
// The order of your output does not matter.


class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> anagrams;
        for (string word : strs)
        {
            string sorted_word = word;
            sort(sorted_word);

            if(map.find(sorted_word) != map.end())
            {
                map[sorted_word].
            }
        }
    }
};